src/frame/utils.js

```diff
@@ -169,14 +176,18 @@ const getTrustedSignals = async (url, keys, debug) => {
 
 	const isJSON = response => /\bapplication\/json\b/.test(response?.headers?.get('content-type'));
 
+	debug && echo.log(echo.asProcess(`fetching keys from trusted signals url: ${url}`));
 	const response = await fetch(`${url}?${hostname}&keys=${keys.join(',')}`)
 		.then(response => {
```

Comment:
seems strange to mix `promise.then` with `await` how about
```
try {
   const response = await fetch(`${url}?${hostname}&keys=${keys.join(',')}`)
} catch {
   // error handling
}
```

Reply:
Good catch.